File: 01a._Data_parsing/parseFiles.js
import fs from "fs/promises";
import yaml from "js-yaml";
import xml2js from "xml2js";

const BASE_DIR = "generated_files/";

// Read JSON
const parseJSON = async (file) => JSON.parse(await fs.readFile(BASE_DIR + file, "utf8"));

// Read YAML
const parseYAML = async (file) => yaml.load(await fs.readFile(BASE_DIR + file, "utf8"));

// Read XML
const parseXML = async (file) => {
  const data = await fs.readFile(BASE_DIR + file, "utf8");
  return xml2js.parseStringPromise(data);
};

// Read CSV
const parseCSV = async (file) => {
  const data = await fs.readFile(BASE_DIR + file, "utf8");
  const [header, ...rows] = data.split("\n").filter(Boolean);
  const keys = header.split(",");
  return rows.map((row) => {
    const values = row.split(",");
    return Object.fromEntries(keys.map((key, i) => [key, values[i]]));
  });
};

// Read TXT
const parseTXT = async (file) => {
  const data = await fs.readFile(BASE_DIR + file, "utf8");
  return data.split("\n").filter(Boolean).map(JSON.parse);
};

// Printing them all out
(async () => {
  console.log("Parsed JSON:", await parseJSON("dataset.json"));
  console.log("------------------------------------");
  console.log("Parsed YAML:", await parseYAML("dataset.yaml"));
  console.log("------------------------------------");
  console.log("Parsed XML:", JSON.stringify(await parseXML("dataset.xml"), null, 2));
  console.log("------------------------------------");
  console.log("Parsed CSV:", await parseCSV("dataset.csv"));
  console.log("------------------------------------");
  console.log("Parsed TXT:", await parseTXT("dataset.txt"));
})();

File: 01a._Data_parsing/parse_files.py
import json
import yaml
import xml.etree.ElementTree as ET
import csv

BASE_DIR = "generated_files/"

# Read JSON
def parse_json(file):
    with open(BASE_DIR + file, "r", encoding="utf-8") as f:
        return json.load(f)

# Read YAML
def parse_yaml(file):
    with open(BASE_DIR + file, "r", encoding="utf-8") as f:
        return yaml.safe_load(f)

# Read XML
def parse_xml(file):
    tree = ET.parse(BASE_DIR + file)
    root = tree.getroot()
    
    def parse_element(element):
        return {child.tag: parse_element(child) if len(child) else child.text for child in element}
    
    return parse_element(root)

# Read CSV
def parse_csv(file):
    with open(BASE_DIR + file, "r", encoding="utf-8") as f:
        reader = csv.DictReader(f)
        return list(reader)

# Read TXT (assuming JSON lines)
def parse_txt(file):
    with open(BASE_DIR + file, "r", encoding="utf-8") as f:
        return [json.loads(line.strip()) for line in f if line.strip()]

# Printing them all out
if __name__ == "__main__":
    print("Parsed JSON:", parse_json("dataset.json"))
    print("------------------------------------")
    print("Parsed YAML:", parse_yaml("dataset.yaml"))
    print("------------------------------------")
    print("Parsed XML:", json.dumps(parse_xml("dataset.xml"), indent=2))
    print("------------------------------------")
    print("Parsed CSV:", parse_csv("dataset.csv"))
    print("------------------------------------")
    print("Parsed TXT:", parse_txt("dataset.txt"))
